opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -187,17 +147,116 @@ where
-    pub fn builder<E, RT>(exporter: E, runtime: RT) -> PeriodicReaderBuilder<E, RT>
+    pub fn builder<E>(exporter: E) -> PeriodicReaderBuilder<E>
+    where
+        E: PushMetricsExporter,
+    {
+        PeriodicReaderBuilder::new(exporter)
+    }
+
+    fn new<E>(exporter: E, interval: Duration) -> Self
     where
         E: PushMetricsExporter,
-        RT: Runtime,
     {
-        PeriodicReaderBuilder::new(exporter, runtime)
+        let (message_sender, message_receiver): (Sender<Message>, Receiver<Message>) =
+            mpsc::channel();
+        let reader = PeriodicReader {
+            exporter: Arc::new(exporter),
+            inner: Arc::new(PeriodicReaderInner {
+                message_sender,
+                is_shutdown: AtomicBool::new(false),
+                producer: Mutex::new(None),
+            }),
+        };
+        let cloned_reader = reader.clone();
+
+        thread::spawn(move || {
+            let mut interval_start = Instant::now();
+            let mut remaining_interval = interval;
+            println!("PeriodicReader Thread started.");
+            loop {
+                match message_receiver.recv_timeout(remaining_interval) {
+                    Ok(Message::Flush(response_sender)) => {
+                        println!("Performing ad-hoc export due to Flush.");
+                        if let Err(_e) = cloned_reader.collect_and_export() {
+                            response_sender.send(false).unwrap();
+                        } else {
+                            response_sender.send(true).unwrap();
+                        }
+
+                        // Adjust the remaining interval after the flush
+                        let elapsed = interval_start.elapsed();
+                        if elapsed < interval {
+                            remaining_interval = interval - elapsed;
+                            println!(
+                                "Adjusting remaining interval after Flush: {:?}",
+                                remaining_interval
+                            );
+                        } else {
+                            // Reset the interval if the flush happens after the expected export time
+                            // This will skip the normal periodic exporting. is this the desired behavior?
+                            interval_start = Instant::now();
+                            remaining_interval = interval;
+                        }
+                    }
+                    Ok(Message::Shutdown(response_sender)) => {
+                        // Perform final export and break out of loop and exit the thread
+                        println!("Performing final export and shutting down.");
+                        if let Err(_e) = cloned_reader.collect_and_export() {
+                            response_sender.send(false).unwrap();
+                        } else {
+                            response_sender.send(true).unwrap();
+                        }
```

Comment:
```suggestion   
response_sender.send(cloned_reader.collect_and_export().is_err())
```

Reply:
i don't plan to keep any on the unwraps. Everything would be handled and logged.